src/screens/Inventory/AddInventory.tsx

```diff
@@ -36,23 +47,45 @@ function AddInventory (props: AddInventoryProps) {
   // Product IDs for items that the site already has inventory for
   const currentSiteProductIds = siteInventory.map((inventory: InventoryRecord) => inventory.productId);
   // Filter to only show products that the site doesn't already have
-  const productOptions = Object.entries(products).filter(([id, _]) => !currentSiteProductIds.includes(id)).map(item => item[0]);
-  
-  const [selectedProductId, setSelectedProductId] = useState("");
+  const productOptionIds = Object.entries(products)
+    .filter(([id, _]) => !currentSiteProductIds.includes(id))
+    .map((item) => item[0]);
+
+  const [selectedProductId, setSelectedProductId] = useState('');
   const [startingAmount, setStartingAmount] = useState(0);
+  const [unit, setUnit] = useState('');
+  const [newProductName, setNewProductName] = useState('');
+  const NEW_PRODUCT_LABEL = '+ New Inventory Item';
 
   // TODO: Add form input validation and error messaging
   const handleSubmit = async (event: React.MouseEvent) => {
     // Prevent page refresh on submit
     event.preventDefault();
+    let productId = selectedProductId; // needed because setSelectedProductId is not immediate
+
+    if (selectedProductId === NEW_PRODUCT_LABEL) {
+      // Create the new product in Airtable and add to Redux
+      const product = JSON.parse(JSON.stringify(EMPTY_PRODUCT));
+      product.unit = unit;
+      product.name = newProductName;
+      delete product.id; // delete id field to add to Airtable
+      try {
+        productId = await createProduct(product);
+      } catch (error) {
+        console.log('[AddInventory] (createProduct, handleSubmit) Error: ', error);
+        productId = generateOfflineId();
+      }
+      product.id = productId;
+      addProductToRedux(product);
+    }
```

Comment:
Aren't we putting this in `request.js`? If so, we should be consistent

Reply:
I moved it into `request.js` and I can update the patch after you check out my (poor) function naming and probably after I merge #66 ?